src/test/java/org/variantsync/vevos/simulation/variability/sequenceextraction/LongestNonOverlappingSequencesTest.java

```diff
@@ -0,0 +1,94 @@
+package org.variantsync.vevos.simulation.variability.sequenceextraction;
+
+import org.junit.Test;
+import org.variantsync.functjonal.list.NonEmptyList;
+import org.variantsync.vevos.simulation.variability.SPLCommit;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class LongestNonOverlappingSequencesTest {
+
+    @Test
+    public void exampleExtraction() {
+        // For example, if the commits comprise three partially overlapping sequences ([A-B-C-D-E], [X-Y-Z], [A-B-F-G]),
+        // the function will return the sequences ([A-B-C-D-E], [X-Y-Z], [F-G]).
+        final SPLCommit a = new SPLCommit("A");
+        final SPLCommit b = new SPLCommit("B");
+        final SPLCommit c = new SPLCommit("C");
+        final SPLCommit d = new SPLCommit("D");
+        final SPLCommit e = new SPLCommit("E");
+        final SPLCommit f = new SPLCommit("F");
+        final SPLCommit g = new SPLCommit("G");
+        final SPLCommit x = new SPLCommit("X");
+        final SPLCommit y = new SPLCommit("Y");
+        final SPLCommit z = new SPLCommit("Z");
+        // A has no parent
+        a.setParents();
+        // A-B
+        b.setParents(a);
+        // B-C
+        c.setParents(b);
+        // C-D
+        d.setParents(c);
+        // D-E
+        e.setParents(d);
+        // B-F
+        f.setParents(b);
+        // F-G
+        g.setParents(f);
+        // X has no parent
+        x.setParents();
+        // X-Y
+        y.setParents(x);
+        // Y-Z
+        z.setParents(y);
+
+        List<SPLCommit> exampleCommits = Arrays.asList(a, b, c, d, e, f, g, x, y, z);
+        NonEmptyList<SPLCommit> expectedOne = new NonEmptyList<>(Arrays.asList(a,b,c,d,e));
+        NonEmptyList<SPLCommit> expectedTwo = new NonEmptyList<>(Arrays.asList(x, y, z));
+        NonEmptyList<SPLCommit> expectedThree = new NonEmptyList<>(Arrays.asList(f, g));
+
+        LongestNonOverlappingSequences algorithm = new LongestNonOverlappingSequences();
+        List<NonEmptyList<SPLCommit>> result = algorithm.extract(exampleCommits);
+
+        assert result.size() == 3;
+        assert sequencesAreEqual(result.get(0), expectedOne);
+        assert sequencesAreEqual(result.get(1), expectedTwo);
+        assert sequencesAreEqual(result.get(2), expectedThree);
```

Comment:
Use JUnit assert methods here? `Assert.assertTrue(...)`? Java keyword `assert` might only work in debug mode.

Reply:
Asserts are always enabled for unit tests afaik. However, I will change it as suggested.

Reply:
Ok, I am not sure. Proceed as you prefer.